modules/microshift-proc-deploying-applications-with-generic-devices.adoc

```diff
@@ -0,0 +1,69 @@
+// Module included in the following assemblies:
+//
+// microshift_configuring/microshift-gdp.adoc
+
+:_mod-docs-content-type: PROCEDURE
+[id="microshift-deploying-applications-with-generic-devices_{context}"]
+= Deploying applications that use generic devices
+
+After the Generic Device Plugin (GDP) is configured and enabled in {microshift-short}, you can deploy Kubernetes workloads, such as pods, deployments, or StatefulSets, that request access to the host devices that you have exposed. The device is made available inside the container without requiring the pod to run with elevated privileges.
+
+.Prerequisites
+
+* You installed {product-title}.
+
+* You enabled and configured GDP.
+
+* You installed {oc-first}.
+
+.Procedure
+
+. Define the device request in your `Pod` specification:
++
+[IMPORTANT]
+====
+In your Kubernetes workload manifest, for example, for a pod, deployment, or StatefulSet, add a `resources.limits` entry to the container that needs device access. The resource name must follow the pattern `device.microshift.io/<device_name>`, where `<device_name>` matches the `name` that you specified in your configuration file. The value indicates the number of instances of that device type the container requires.
+====
++
+[source,yaml]
+----
+apiVersion: v1
+kind: Pod
+metadata:
+  name: device-app
+spec:
+  containers:
+  - name: container
+    image: <your_application_image> <1>
+    command: ["/path/to/your/app"] <2>
+    args: ["--device_path=/dev/video0"] <3>
+    resources:
+      limits:
+        device.microshift.io/video: 1 <4>
+----
```

Comment:
To prevent the container from gaining unwanted access or elevated privileges beyond what's absolutely necessary, you should explicitly define a securityContext and configure it with the principle of least privilege in mind. This ensures the container only has the permissions it needs, like access to the device file, while its other capabilities remain restricted." so pod definition should be included to also add below content.

```
 resources:
          limits:
            device.microshift.io/fakeserial: 1
        securityContext:
          allowPrivilegeEscalation: false
          capabilities:
            drop: ["ALL"]
          runAsNonRoot: true
          seccompProfile:
            type: "RuntimeDefault"
```